feature/1: initial implementation (#2)
exasol/autogenerated-resource-verifier-java: README.md

# Autogenerated Resource Verifier for Java

[![Maven Central](https://img.shields.io/maven-central/v/com.exasol/autogenerated-resource-verifier-java)](https://search.maven.org/artifact/com.exasol/autogenerated-resource-verifier-java)

> This Java library allows you to define autogenerated resources in Java.

Reviewer: ```suggestion
This Java library allows you to verify autogenerated resources in Java.
```

Reviewer: Did you meant `verify` here? If so, maybe add one more sentence what we are verifying.

Reviewer: After look the code, maybe `create` is better here.

```
This Java library allows you to automatically create resources in Java.
```

Author: ```suggestion
This Java library allows you to automatically create resources in Java.
```

## Usage

Define the content of your document in your test source code:

```java
public class MyReportGenerator {

    public static void main(final String[] args) {

>         new AutogeneratedResourceVerifier().verifyResource(Path.of(args[0]), new CapabilitiesReport());

Reviewer: ```suggestion
        new AutogeneratedResourceVerifier().verifyResource(Path.of(args[0]), new MyReport());
```

            return Path.of("doc", "generated", "myReport.md");
        }
    }
}
```

### Run From Maven:

You can add the execution of your generator to the maven build using the `maven-execution-plugin`. To do so add the following lines to your `pom.xml`:

```xml

<plugin>
    <groupId>org.codehaus.mojo</groupId>
    <artifactId>exec-maven-plugin</artifactId>
    <version>3.0.0</version>
    <executions>
        <execution>
            <id>validate-capabilities-report</id>
            <goals>
                <goal>java</goal>
            </goals>
            <phase>package</phase>
            <configuration>
                <mainClass>com.example.MyReportGenerator</mainClass> <!-- Add your class here -->
                <classpathScope>test</classpathScope>
                <systemProperties>
                    <systemProperty>
                        <key>projectDir</key>
                        <value>${project.basedir}</value>
                    </systemProperty>
                </systemProperties>
            </configuration>
        </execution>
    </executions>
</plugin>
```

Now you can run the validation using `mvn package`.

> The verifier can also fix (override) the generated resource. To do so, run:

Reviewer: ```suggestion
The verifier can also fix the generated resource. To do so, run:
```

```shell
mvn package -DfixAutogeneratedResources=true
```

### Run Time Dependencies

> Running the Virtual Schema requires a Java Runtime version 11 or later.

Reviewer: ```suggestion
Running the project requires a Java Runtime version 11 or later.
```

| Dependency                                                         | Purpose                                                | License                       |
|--------------------------------------------------------------------|--------------------------------------------------------|-------------------------------|

> | [Exasol Error Reporting][error-reporting-java]                     | Common JDBC functions for Virtual Schemas adapters     | MIT License                   |

Reviewer: ```suggestion
| [Exasol Error Reporting][error-reporting-java]                     | Java builders for Exasol error messages.     | MIT License                   |
```

[maven-assembly-plugin]: https://maven.apache.org/plugins/maven-assembly-plugin/

[maven-failsafe-plugin]: https://maven.apache.org/surefire/maven-failsafe-plugin/

[sonatype-oss-index-maven-plugin]: https://sonatype.github.io/ossindex-maven/maven-plugin/

[versions-maven-plugin]: https://www.mojohaus.org/versions-maven-plugin/

[maven-enforcer-plugin]: http://maven.apache.org/enforcer/maven-enforcer-plugin/

[artifact-ref-checker-plugin]: https://github.com/exasol/artifact-reference-checker-maven-plugin

[project-keeper-maven-plugin]: https://github.com/exasol/project-keeper-maven-plugin

[postgresql-jdbc-driver]: https://jdbc.postgresql.org/

[test-db-builder]: https://github.com/exasol/test-db-builder/

[virtual-schema-common-jdbc]: https://github.com/exasol/virtual-schema-common-jdbc

[exasol-testcontainers]: https://github.com/exasol/exasol-testcontainers

[user-guide]: https://docs.exasol.com/database_concepts/virtual_schemas.htm

[virtual-schemas]: https://github.com/exasol/virtual-schemas

[vs-api]: https://github.com/exasol/virtual-schema-common-java/blob/master/doc/development/api/virtual_schema_api.md

[vs-doc]: https://github.com/exasol/virtual-schemas/tree/master/doc

> 

Reviewer: ```suggestion
```
